packages/lesswrong/lib/collections/comments/schema.ts

```diff
@@ -395,6 +414,288 @@ const schema: SchemaType<DbComment> = {
+      type: "User",
+      nullable: true,
+    }),
+    optional: true,
+    canRead: ['guests'],
+    canUpdate: ['sunshineRegiment', 'admins'],
+    canCreate: ['members'],
+    hidden: true,
+    onEdit: (modifier, document, currentUser) => {
+      if (modifier.$set && (modifier.$set.deletedPublic || modifier.$set.deleted) && currentUser) {
+        return modifier.$set.deletedByUserId || currentUser._id
+      }
+    },
+  },
+
+  // spam: Indicates whether a comment has been marked as spam.
+  // This removes the content of the comment, but still renders replies.
+  spam: {
+    type: Boolean,
+    optional: true,
+    canRead: ['guests'],
+    canCreate: ['admins'],
+    canUpdate: ['admins'],
+    control: "checkbox",
+    hidden: true,
+    ...schemaDefaultValue(false),
+  },
+
+  // repliesBlockedUntil: Deactivates replying to this post by anyone except
+  // admins and sunshineRegiment members until the specified time is reached.
+  repliesBlockedUntil: {
+    type: Date,
+    optional: true,
+    group: moderationOptionsGroup,
+    canRead: ['guests'],
+    canUpdate: ['sunshineRegiment', 'admins'],
+    control: 'datetime'
+  },
+
+  needsReview: {
+    type: Boolean,
+    optional: true,
+    canRead: ['guests'],
+    canUpdate: ['sunshineRegiment', 'admins'],
+    canCreate: ['sunshineRegiment', 'admins'],
+    hidden: true,
+  },
+
+  reviewedByUserId: {
+    ...foreignKeyField({
+      idFieldName: "reviewedByUserId",
+      resolverName: "reviewedByUser",
+      collectionName: "Users",
+      type: "User",
+      nullable: true,
+    }),
+    optional: true,
+    canRead: ['guests'],
+    canUpdate: ['sunshineRegiment', 'admins'],
+    canCreate: ['sunshineRegiment', 'admins'],
+    hidden: true,
+  },
+
+  // hideAuthor: Displays the author as '[deleted]'. We use this to copy over
+  // old deleted comments from LW 1.0
+  hideAuthor: {
+    type: Boolean,
+    group: moderationOptionsGroup,
+    optional: true,
+    canRead: ['guests'],
+    canUpdate: ['admins'],
+    ...schemaDefaultValue(false),
+  },
+  
+  moderatorHat: {
+    type: Boolean,
+    optional: true,
+    canRead: ['guests'],
+    canUpdate: ['sunshineRegiment', 'admins'],
+    canCreate: ['sunshineRegiment', 'admins'],
+    ...schemaDefaultValue(false),
+    hidden: true,
+  },
+
+  // whether this comment is pinned on the author's profile
+  isPinnedOnProfile: {
+    type: Boolean,
+    optional: true,
+    canRead: ['guests'],
+    canCreate: ['members'],
+    canUpdate: [userOwns, 'sunshineRegiment', 'admins'],
+    hidden: true,
+    ...schemaDefaultValue(false),
+  },
+  
+  /* Alignment Forum fields */
+  af: {
+    type: Boolean,
+    optional: true,
+    label: "AI Alignment Forum",
+    ...schemaDefaultValue(false),
+    viewableBy: ['guests'],
+    editableBy: ['alignmentForum', 'admins'],
+    insertableBy: ['alignmentForum', 'admins'],
+    hidden: (props) => alignmentForum || !props.alignmentForumPost
+  },
+
+  afBaseScore: {
```

Comment:
I agree with moving the AF fields to the same file for simplicity (when I first created them i think we hadn't thoroughly moved away from Vulcan yet and I was trying to preserve some kind of modularity that by now we've thoroughly given up on)

I think it might be nice to have a little separation between AF and regular fields, for the benefit of future Forum Magnum developers setting up their own instance that are like "wtf are these AF fields?".

Maybe, like, use "addFieldsDict" the way custom fields was previously doing, but rather than putting it in a separate file, just put it in the same schema file (so we can see that they're a separate group of fields that will reliably stay separate as we add more fields later, but, we don't have to guess which file the fields live in)

Reply:
Unfortunately `addFieldsDict` relies on `collection.ts`, which creates a dependency cycle. I could use `Object.assign`, if you'd prefer that?

Reply:
(Went ahead and added separation using `Object.assign` in 97e327a97bfed2fdb016a3d886101b2d366c9551)

Reply:
Yeah that sounds good.